src/interpolation_transfer.jl

```diff
@@ -169,6 +166,35 @@ function transfer_blochwave_kpt(ψk_in, basis_in::PlaneWaveBasis{T}, kpt_in::Kpo
     ψk_out
 end
 
+"""
+Shift an object `x` by `y` with modulus of `l`.
+"""
+function mod1_shift(x::T, shift, l) where T
+    T([mod1(x[i] + shift[i], l[i]) for i in 1:length(l)]...)
+end
+
+"""
+Transfer an array ψk defined on basis ``k``-point kpt_in to shifted ``k``-point kpt_out.
+Useful for phonons.
+"""
+# TODO: look for a lossless way to do this
+function transfer_blochwave_kpt(ψk_in, basis::PlaneWaveBasis{T}, kpt_in::Kpoint,
+                                kpt_out::Kpoint, shift) where T
+    ψk_out = transfer_blochwave_kpt(ψk_in, basis, kpt_in, basis, kpt_out)
+
+    ind_G_vecs = index_G_vectors.(Ref(basis), G_vectors(basis))
+    shifted_ind_G_vecs = mod1_shift.(ind_G_vecs, Ref(shift), Ref(size(ind_G_vecs)))
+
+    idcs_out = index_G_vectors.(Ref(basis), G_vectors(basis)[shifted_ind_G_vecs])
+    idcs_out = getindex.(Ref(LinearIndices(basis.fft_size)), idcs_out)
+    idcs_out = indexin(idcs_out, kpt_out.mapping)
+    if any(isnothing, idcs_out)
+        idcs_out = idcs_out[idcs_out .!= nothing]
```

Comment:
again weird logic due to broadcast. Who gave you such a traumatic experience, python or matlab? :-p